src/plugins/hiddenColumns/hiddenColumns.js

```diff
@@ -153,33 +153,45 @@ class HiddenColumns extends BasePlugin {
    */
   showColumns(columns) {
     const currentHideConfig = this.getHiddenColumns();
-    const isConfigValid = this.isValidConfig(columns);
+    const isValidConfig = this.isValidConfig(columns);
     let destinationHideConfig = currentHideConfig;
+    const hidingMapValues = this.#hiddenColumnsMap.getValues().slice();
+    const isAnyColumnShowed = columns.length > 0;
 
-    if (isConfigValid) {
-      destinationHideConfig = currentHideConfig.filter(column => columns.includes(column) === false);
+    if (isValidConfig && isAnyColumnShowed) {
+      const physicalColumns = columns.map(visualColumn => this.hot.toPhysicalColumn(visualColumn));
+
+      // Preparing new values for hiding map.
+      arrayEach(physicalColumns, (physicalColumn) => {
+        hidingMapValues[physicalColumn] = false;
+      });
+
+      // Preparing new hiding config.
+      destinationHideConfig = arrayReduce(hidingMapValues, (hiddenIndexes, isHidden, physicalIndex) => {
+        if (isHidden) {
+          hiddenIndexes.concat(this.hot.toVisualColumn(physicalIndex));
```

Comment:
```suggestion
          hiddenIndexes.push(this.hot.toVisualColumn(physicalIndex));
```
The `concat` method doesn't have any effect on the `destinationHideConfig` array. Hence, the hook `beforeUnhideColumns` gets an empty array while showing the columns. Does it seem that there are no tests for those arguments?

Reply:
Huh, it looks like it! Thanks for mindfulness. I will add extra test.